.github/workflows/tests.yaml

```diff
@@ -0,0 +1,89 @@
+name: Test
+
+permissions:
+  contents: read
+
+on: [push, pull_request]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: ${{ startsWith(github.ref, 'refs/pull/') }}
+
+jobs:
+  test:
+    name: "Tests (${{ matrix.compiler }}, C++${{ matrix.standard}}, ${{ matrix.mode }}, ${{ matrix.enables }})"
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        # only the compilers supported by setup-cpp
+        compiler: [clang++-18, gcc-13]
+        standard: [20, 23]
+        mode: [dev, debug, release]
+        include:
+          # in addition to the above combinations, add a combination to build with
+          # dpdk. but only build this combination when dpdk is enabled, so we don't
```

Comment:
Nitpick (not important now, I'll merge it as-is) I found the phrase "only build this combination when dpdk is enable" confusing - the phrase "in addition to the above" says the opposite, that we did build this combination without dpdk but now we'll build it again with dpdk.

Reply:
thank you @nyh . i sent another PR to address this. HTH.